docs: improve README with complete examples, dependencies, and troubl…

Ruthwik000 opened the pull request:
issue:#4
I had added:
Dependencies section - Install commands for Ubuntu, Fedora, Arch, and macOS
Fixed Examples - Complete command paths for all 4 examples (were truncated)
Troubleshooting section - Solutions for common build/runtime errors
into the readme

Ruthwik000 committed:
docs: improve README with complete examples, dependencies, and troubleshooting

Ruthwik000 commented:
@chimosky  kindly review and if you find it useful kindly merge it

chimosky commented on README.md:
````diff
@@ -2,6 +2,36 @@
+### macOS
+```bash
+brew install meson ninja gtk4 glib
+```
````
I honestly don't see the point of having these installation instructions here, I'd rather list the dependencies and have them figure out how to install it.

It's a basic challenge that anyone building this should be able to do.

chimosky commented on README.md:
```diff
@@ -34,19 +64,62 @@ Tests include:
+# File attributes example
 ./builddir/examples/sugar_file_attributes_example
+
+# Long press controller example
```
These comments aren't really helpful as the file names tell what kinda example they are, please remove them.

chimosky commented on README.md:
````diff
@@ -34,19 +64,62 @@ Tests include:
+## Troubleshooting
+
+### Meson not found
+If you get `meson: command not found`, install Meson using your package manager or pip:
+```bash
+pip install --user meson ninja
````
Another part that's not needed, because we've stated the dependencies above.

chimosky commented on README.md:
````diff
@@ -34,19 +64,62 @@ Tests include:
+### GTK4 version incompatibility
+Ensure you have GTK4 >= 4.0 installed:
+```bash
+pkg-config --modversion gtk4
````
Another part that's not needed.

chimosky commented on README.md:
````diff
@@ -34,19 +64,62 @@ Tests include:
+If you encounter build errors, try cleaning and rebuilding:
+```bash
+rm -rf builddir
+meson setup builddir
````
Meson has a `--wipe` argument that's neater and cleaner and achieves the same result without using `rm`.

Ruthwik000 committed:
docs: simplify README based on review feedback

Ruthwik000 commented:
@chimosky , I addressed all  comments:
 Removed platform-specific install commands - just listing dependencies now
Removed comments from Examples section (# Grid layout example, # Event controller example, etc.)
Removed GTK4 version check code (pkg-config --modversion gtk4)
Changed rm -rf builddir to meson setup builddir --wipe as suggested

kindly review

chimosky commented:
Thank you!
